### PR TITLE
Optuna hyperparameter optimization tutorial

### PR DESCRIPTION
**Pull request by TZTsai**
Updated distributions in "distribution.py", added an optuna util file and Tutorial 8 on using optuna for HPO.

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/vanderschaarlab/.github/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/vanderschaarlab/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [van der Schaar Lab Styleguide](https://github.com/vanderschaarlab/.github/blob/main/STYLEGUIDE.md)
- [X] I have labelled this PR with the relevant [Type labels](https://github.com/vanderschaarlab/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

**Review by bcebere — CHANGES_REQUESTED** [review]
Thank you for your contribution!

There are a few more rules to follow for notebooks:
 - save the clean `ipynb`, without any generated output
 - could you add by default adsgan/ctgan, instead of nflows? And try that? from logs, it looks that there are a bunch of weird fials
 
 @robsdavis Please guide @TZTsai to wrap this PR.
 
 Thanks!

**Comment by TZTsai** [comment]
> Thank you for your contribution!
> 
> There are a few more rules to follow for notebooks:
> 
>     * save the clean `ipynb`, without any generated output
> 
>     * could you add by default adsgan/ctgan, instead of nflows? And try that? from logs, it looks that there are a bunch of weird fials
> 
> 
> @robsdavis Please guide @TZTsai to wrap this PR.
> 
> Thanks!

Hi @bcebere! I tried ctgan but some combinations of hyperparameters led to out-of-memory errors. I think it is normal to have invalid combinations of hyperparameters because the hyperparameter space cannot include more fine-grained constraints like `param_a < param_b` etc. I have changed to tvae by default for now. Please review the update.

**Comment by robsdavis** [comment]
Hi @TZTsai, Thanks for all your work on this. I'll do a detailed review, if you get the test to pass again. You have a mypy error due to a re-definition of `offset`. On lines 382/386. Thanks

**Comment by TZTsai** [comment]
@bcebere @robsdavis There seems to be a dependency issue when testing. It failed to import torch. Could you help check it?

**Comment by HLasse** [comment]
Thanks a lot for this - it's very timely :) is there an ETA for when this will get merged? Would be extremely useful for me

**Review by robsdavis — APPROVED** [review]

**Comment by robsdavis** [comment]
Thanks @TZTsai for another great contribution!
